src/Eris/Generator/SuchThatGenerator.php

```diff
@@ -69,6 +76,17 @@ public function contains(GeneratedValue $value)
 
     private function predicate(GeneratedValue $value)
     {
-        return call_user_func($this->filter, $value->unbox());
+        if ($this->filter instanceof PHPUnit_Framework_Constraint) {
+            try {
+                $this->filter->evaluate($value->unbox());
+                return true;
+            } catch (PHPUnit_Framework_ExpectationFailedException $e) {
+                return false;
+            }
+        } else if (is_callable($this->filter)) {
```

Comment:
@giorgiosironi 

Since we always return early here, how about changing this to

```php
if ($this->filter instanceof PHPUnit_Framework_Constraint) {
    try {
        $this->filter->evaluate($value->unbox());
    } catch (PHPUnit_Framework_ExpectationFailedException $e) {
        return false;
     }

    return true;
} 

if (is_callable($this->filter)) {
    return call_user_func($this->filter, $value->unbox());
}

throw new LogicException("Specified filter does not seem to be of the correct type. Please pass a callable or a PHPUnit_Framework_Constraint instead of " . var_export($this->filter, true));
```